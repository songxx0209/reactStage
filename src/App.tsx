import React from 'react';
import AddTodo from './components/AddTodo';
import TodoList from './components/TodoList';
// import Ant from './components/Ant';

interface Props {}

interface State {
  list: Array<any>,
}

export default class App extends React.Component<{}, State> {
  constructor(props: any) {
    super(props);
    this.state = {
      list: [],
    }
  }
  addItem(value: any) {
    let newState = this.state.list;
    newState.push({
      value,
      id: new Date().getTime(),
    });
    this.setState({
      list: newState,
    });
  }
  delItem(id: any) {
    const { list } = this.state;
    const newState = list.filter(item => {
      return item.id !== id;
    });

    console.log('newState=', newState);
    this.setState({
      list: newState
    });
  }

  render() {
    console.log(this.state.list);
    return (
      <div>
        <h1>Todos</h1>
        <TodoList itemList={this.state.list} delItem={this.delItem.bind(this)} />
        <AddTodo addItem={this.addItem.bind(this)} />
        <hr/>
        {/* <Ant name="song" company={`woqu`}></Ant> */}
      </div>
    );
  }
}