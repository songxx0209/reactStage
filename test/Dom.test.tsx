import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App';


// 快照测试
describe('<App /> ', () => {
  it('Snapshot', () => {
    const component = renderer.create(<App/>);
    let snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});

