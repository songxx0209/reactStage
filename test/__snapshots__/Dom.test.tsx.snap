// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App />  Snapshot 1`] = `
<div>
  <h1>
    Todos
  </h1>
  <ul />
  <div
    className="add-todo"
  >
    <input
      onChange={[Function]}
      placeholder="Todo four"
      type="text"
      value=""
    />
    <button
      className="add-button"
      onClick={[Function]}
    >
      Add Todo
    </button>
  </div>
  <hr />
</div>
`;
